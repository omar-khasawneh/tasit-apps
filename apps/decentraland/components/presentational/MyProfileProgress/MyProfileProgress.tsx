import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import { StyleSheet, Text, View } from "react-native";
import React from "react";
import Colors from "../../../constants/Colors";
import ProgressBar from "react-native-progress/Bar";

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-start",
  },
  progressText: {
    color: Colors.textColor,
    fontWeight: "800",
  },
  progressTextContainer: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: responsiveHeight(1),
  },
});

interface MyProfileProgressProps {
  progress: number;
}

export const MyProfileProgress: React.FunctionComponent<MyProfileProgressProps> = React.memo(
  ({ progress }) => {
    return (
      <View style={styles.container}>
        <ProgressBar
          borderRadius={0}
          borderWidth={0}
          color={Colors.loadingColor}
          height={responsiveHeight(1)}
          progress={progress}
          unfilledColor={Colors.textColor}
          width={responsiveWidth(100)}
        />
        <MyProfileProgressText progress={progress} />
      </View>
    );
  }
);

interface MyProfileProgressTextProps {
  progress: number;
}

export const MyProfileProgressText: React.FunctionComponent<MyProfileProgressTextProps> = React.memo(
  ({ progress }) => {
    const formattedProgress = Math.round(progress * 100);

    return (
      <View style={styles.progressTextContainer}>
        <Text style={styles.progressText}>{formattedProgress}% complete</Text>
      </View>
    );
  }
);

export default MyProfileProgress;
