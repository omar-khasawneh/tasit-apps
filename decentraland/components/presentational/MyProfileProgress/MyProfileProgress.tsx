import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import { StyleSheet, Text, View } from "react-native";
import React from "react";
import Colors from "../../../constants/Colors";
import ProgressBar from "react-native-progress/Bar";

interface MyProfileProgressProps {
  progress: number;
}

export const MyProfileProgress: React.SFC<MyProfileProgressProps> = ({
  progress,
}) => {
  return (
    <View style={styles.container}>
      <ProgressBar
        borderRadius={0}
        borderWidth={0}
        color={Colors.loadingColor}
        height={responsiveHeight(1)}
        progress={progress}
        unfilledColor={Colors.textColor}
        width={responsiveWidth(100)}
      />
      <MyProfileProgressText progress={progress} />
    </View>
  );
};

interface MyProfileProgressTextProps {
  progress: number;
}

export const MyProfileProgressText: React.SFC<MyProfileProgressTextProps> = ({
  progress,
}) => {
  const formattedProgress = Math.round(progress * 100);

  return (
    <View style={styles.progressTextContainer}>
      <Text style={styles.progressText}>{formattedProgress}% complete</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-start",
  },
  progressText: {
    color: Colors.textColor,
    fontWeight: "800",
  },
  progressTextContainer: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: responsiveHeight(1),
  },
});

export default MyProfileProgress;
