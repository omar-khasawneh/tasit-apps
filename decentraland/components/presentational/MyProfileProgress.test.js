import React from "react";
import { shallow } from "enzyme";
import { MyProfileProgress, MyProfileProgressText } from "./MyProfileProgress";

describe("MyProfileProgress", () => {
  it("renders the component", async () => {
    const progress = 0.5;
    expect(
      shallow(<MyProfileProgress progress={progress} />)
    ).toMatchSnapshot();
  });

  describe("MyAccountProgressText", () => {
    it("renders the component", async () => {
      const progress = 0.5;
      expect(
        shallow(<MyProfileProgressText progress={progress} />)
      ).toMatchSnapshot();
    });
  });
});
