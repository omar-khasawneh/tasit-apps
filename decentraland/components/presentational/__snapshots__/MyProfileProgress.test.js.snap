// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfileProgress MyAccountProgressText renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginTop": 13.34,
    }
  }
>
  <Component
    style={
      Object {
        "fontWeight": "800",
      }
    }
  >
    50
    % complete
  </Component>
</Component>
`;

exports[`MyProfileProgress renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={0}
    borderWidth={0}
    color="#2196f3"
    height={13.34}
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    progress={0.5}
    unfilledColor="gray"
    useNativeDriver={false}
    width={750}
  />
  <MyProfileProgressText
    progress={0.5}
  />
</Component>
`;
